### PR TITLE
The escape() function is deprecated.

### LINKED ISSUE #22034: Keycloak, javascript lib to not use the escape() function
### Description

escape() function is deprecated and no longer recommended to use.
[MDN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape)

### Discussion

_No response_

### Motivation

_No response_

### Details

The current escape() method is used to decode the jwt-token obtained from the server and extract the information from access token and id token.
Replace this with a more standard library based decoding like `jwt-decode`

Created a PR for this - [21990](https://github.com/keycloak/keycloak/pull/21990)

### PR DESCRIPTION
**Pull request by Abhishek-Govula**
Closes #22034

### CODE REVIEW

File: js/libs/keycloak-js/src/keycloak.js [hunk 1]
```diff
@@ -1021,27 +1022,11 @@ function Keycloak (config) {
     }
 
     function decodeToken(str) {
```
Comment (jonkoops) [inline js/libs/keycloak-js/src/keycloak.js:1024]: I think we can replace this entire function with `jwtDecode`, there should be no need to keep the exception the same here.

File: js/libs/keycloak-js/package.json
Comment (jonkoops) [inline js/libs/keycloak-js/package.json:1]: Can you also check in the changes to the `pnpm-lock.yaml` file?

**Review by jonkoops — CHANGES_REQUESTED** [review]
Thanks for this @Abhishek-Govula! Just have a couple of small notes.

**Review by Abhishek-Govula — COMMENTED** [review]
Updated the changes as requested on decodeToken. Also added the pnpm dependency

File: js/libs/keycloak-js/package.json
Comment (Abhishek-Govula) [reply js/libs/keycloak-js/package.json:1]: Done with the changes. Can you please review

**Review by jonkoops — APPROVED** [review]
LGTM, let's see if the CI passes.

**Comment by jonkoops** [comment]
@Abhishek-Govula could you log an issue for this and link it to this PR? We need it for our release notes.

**Comment by Abhishek-Govula** [comment]
@jonkoops - Create the issue and linked the PR in the description [Issue](https://github.com/keycloak/keycloak/issues/22034)

**Comment by Abhishek-Govula** [comment]
Thanks for the merge @jonkoops 👍

**Pull request merged**